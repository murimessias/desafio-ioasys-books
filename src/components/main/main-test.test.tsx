import { render, screen } from '@testing-library/react'
import { Main } from '.'

const props = {
  title: 'Desafio Ioasys',
  description: 'Boilerplate para o desafio da Ioasys',
}

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main {...props} />)

    expect(
      screen.getByRole('heading', { name: /desafio ioasys/i }),
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('shoud render the backgroun color', () => {
    const { container } = render(<Main {...props} />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#111827' })
  })
})
