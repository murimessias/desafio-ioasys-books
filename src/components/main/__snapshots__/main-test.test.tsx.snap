// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main>
  <h1>
    Desafio Ioasys
  </h1>
</main>
`
